go.mod

```diff
@@ -1,25 +1,28 @@
 module open-cluster-management.io/cluster-proxy
 
-go 1.20
+go 1.21
+
+toolchain go1.22.0
```

Comment:
How about seperate an independent PR for just upgrade go version to 1.22?

Reply:
https://github.com/open-cluster-management-io/cluster-proxy/pull/176

Once merged, I can rebase this pr